service/src/main/java/bio/terra/workspace/service/workspace/WsmApplicationService.java

```diff
@@ -119,7 +120,16 @@ public WsmWorkspaceApplication launchApplicationAbleJobAndWait(
 
   public WsmWorkspaceApplication getWorkspaceApplication(
       Workspace workspace, String applicationId) {
-    return applicationDao.getWorkspaceApplication(workspace.getWorkspaceId(), applicationId);
+    return getWorkspaceApplicationByWorkspaceId(workspace.getWorkspaceId(), applicationId);
+  }
+
+  public WsmWorkspaceApplication getWorkspaceApplicationByWorkspaceId(
```

Comment:
Why have 2 methods that do the same thing?

Reply:
Good catch, fixed.